src/lib.rs

```diff
@@ -115,7 +123,86 @@ impl Procyon {
+        let mut webdriver_process = Command::new("geckodriver")
+            .args(&["--port", "4444"])
+            .spawn()
+            .expect("webdriver failed to start");
+        // This should also be TODO:updated to all webdriver
+        let mut caps = serde_json::map::Map::new();
+        let opts = serde_json::json!({ "args": ["--headless"]});
+        caps.insert("moz:firefoxOptions".to_string(), opts);
+
+        let mut connection_ok = false;
+        while !connection_ok {
+            thread::sleep(time::Duration::from_secs(5));
+            // This should also be TODO:updated to all webdriver
+            let mut caps = serde_json::map::Map::new();
+            let opts = serde_json::json!({ "args": ["--headless"]});
+            caps.insert("moz:firefoxOptions".to_string(), opts);
+            let mut connection_ok = Client::with_capabilities("http://localhost:4444", caps)
+                .await
+                .is_err();
+            if connection_ok == false {
+                break;
+            };
+        }
+        let mut c = Client::with_capabilities("http://localhost:4444", caps).await?;
+        // let mut c = retry(Fixed::from_millis(100), || {
+        //     match Some(c.as_ref().unwrap()) {
+        //         Some(Client { tx, is_legacy }) => OperationResult::Ok(c),
+        //         _ => OperationResult::Retry("not connected yet"),
+        //     }
+        // })
+        // .unwrap()?;
+
+        //.expect("failed to connect to WebDriver");
+
+        // Current implem use the saved html by showata and imitate user clik to save the image
+        // Another approach is to updated the embeded js like in altair:
+        // https://github.com/altair-viz/altair_saver/blob/master/altair_saver/savers/_selenium.py
+        dbg!(self.build().to_html_file()?.to_str().unwrap());
+        c.goto(&format!(
+            "file:///private{}",
+            self.build().to_html_file()?.to_str().unwrap()
+        ))
+        .await?;
+        // c.goto("file:///private/var/folders/32/h6lt_67s75g6jf3h4hx_myxc0000gn/T/showata/show-1587655243189631000.html").await?;
+        let mut summary_button = c
+            .wait_for_find(Locator::Css("summary"))
+            .await?
+            .click()
+            .await?;
+        let mut hidden_link = summary_button
+            .find(Locator::LinkText("Save as PNG"))
+            .await?
+            .click()
+            .await?;
+
+        let link = hidden_link
+            .find(Locator::LinkText("Save as PNG"))
+            .await?
+            .attr("href")
+            .await?;
+
+        let image_data_url = link.unwrap();
+        //let data64: Vec<&str> = image_data_url.split(',').collect::<Vec<&str>>();
+        //dbg!(data64[1]);
+        //let bytes: Vec<u8> = base64::decode(data64[1]).unwrap();
+        // "png;base64,iVB"
+        let _format = &image_data_url[11..14];
+        let bytes: Vec<u8> = base64::decode(&image_data_url[22..]).unwrap();
```

Comment:
not sure if it's worth the cost, but there is a crate to parse data url : https://crates.io/crates/data-url
instead of magic numbers...

Reply:
you don't like magic numbers 😄 I will look into it ans see if i can help for `svg`format

Reply:
no need to introduce a crate just for that, but you can replace the magic number by the "formula" used to compute eg: something like `"data:".length`

Reply:
I saw you comment a bit late. Current version of the code use DataUrl but we can discuss about it.